datafusion/physical-expr/src/aggregate/min_max.rs

```diff
@@ -1103,3 +1117,36 @@ impl Accumulator for SlidingMinAccumulator {
         std::mem::size_of_val(self) - std::mem::size_of_val(&self.min) + self.min.size()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn float_max_with_nans() {
+        let pos_nan = f32::NAN;
+        let zero = 0_f32;
+
+        let vals_float = Float32Array::from_iter_values([zero, pos_nan].iter().copied());
+
+        let vals: ArrayRef = Arc::new(vals_float.clone());
+
+        // We accumulate the above two rows in two different ways.  First, we pass in both as a single batch
+        let mut accumulator = MaxAccumulator::try_new(&DataType::Float32).unwrap();
+        accumulator.update_batch(&[vals]).unwrap();
+        let single_batch_result = &accumulator.evaluate().unwrap();
+
+        // Next we pass the two values in two different batches.
+        let vals_a: ArrayRef =
+            Arc::new(Float32Array::from_iter_values([pos_nan].iter().copied()));
+        let vals_b: ArrayRef =
+            Arc::new(Float32Array::from_iter_values([zero].iter().copied()));
+
+        let mut accumulator = MaxAccumulator::try_new(&DataType::Float32).unwrap();
+        accumulator.update_batch(&[vals_a]).unwrap();
+        accumulator.update_batch(&[vals_b]).unwrap();
+        let split_batch_result = &accumulator.evaluate().unwrap();
+
+        assert_eq!(single_batch_result, split_batch_result);
```

Comment:
I think it would be valuable to also test:
1. `min` 
2. the actul value of `single_batch_result` (to show if it is expected to produce nan or 0)